### PR TITLE
Support for oneOf

### PR DESCRIPTION
**Pull request by 0181532686cf4a31163be0bf3e6bb6732bf**
Rebased @maroux work (#132)

### CODE REVIEW

File: flex/utils.py [hunk 1]
```diff
@@ -199,3 +199,14 @@ def dereference_reference(reference, context):
...
+
+def exactly_one(iterable):
+    """
+    Returns True if bool(x) is True for exactly one x in the iterable
+    If the iterable is empty, returns False
+    """
+    it = iter(iterable)
+    # check that one is True, then check that none is True in rest of the
+    # iterator
+    return any(it) and not any(it)
```
Comment (lyssdod) [inline flex/utils.py]: @maroux @pipermerriam can you advise me on the missing tests for this?

Comment (pipermerriam) [reply flex/utils.py]: This should be tested that for any iterable passed into it:

- returns `True` if **exactly** one item from the iterable evaluates *truthy*
- returns `False` if:
  - None of the elements evaluate *truthy*
  - More than one element evaluates *truthy*
  - The iterable is empty.

**Comment by pipermerriam** [comment]
@lyssdod looks like there are some linting errors.  You can see them here: https://travis-ci.org/pipermerriam/flex/jobs/306744346

Or run the linter yourself locally with `tox flake8` or `flake flex`

**Comment by fruch** [comment]
isn't one of part of OAS 3.0 ?

**Pull request closed without merging**